lib/jsdom/living/helpers/http-request.js

```diff
@@ -0,0 +1,272 @@
+"use strict";
+const http = require("http");
+const https = require("https");
+const { Writable } = require("stream");
+const zlib = require("zlib");
+
+const ver = process.version.replace("v", "").split(".");
+const majorNodeVersion = Number.parseInt(ver[0]);
+const minorNodeVersion = Number.parseInt(ver[1]);
+
+function abortRequest(clientRequest) {
+  if (majorNodeVersion > 13 || (majorNodeVersion === 13 && minorNodeVersion > 13)) {
+    clientRequest.destroy();
+  } else {
+    clientRequest.abort();
+  }
+  clientRequest.removeAllListeners();
+  clientRequest.on("error", () => {});
+}
+
+class Request extends Writable {
+  constructor(uriHref, options) {
+    super();
+    this._currentUrl = uriHref;
+    this._options = options;
+    this.headers = this._options.headers;
+    this._ended = false;
+    this._ending = false;
+    this._redirectCount = 0;
+    this._requestBodyLength = 0;
+    this._requestBodyBuffers = [];
+    this.response = Object.create(null);
+    this._performRequest();
+  }
+
+  abort() {
+    abortRequest(this._currentRequest);
+    this.emit("abort");
+    this.removeAllListeners();
+  }
+
+  getCookie() {
+    let cookies;
+    if (this._redirectCount === 0) {
+      this.originalCookieHeader = this.getHeader("Cookie");
+    }
+    const jar = this._options.cookieJar;
+    if (jar && typeof jar.getCookieStringSync === "function") {
+      cookies = jar.getCookieStringSync(this._currentUrl);
+    } else {
+      this._areCookiesDisabled = true;
```

Comment:
It seems like this shouldn't be set every time you call getCookie, but instead should be determined in the constructor.